### PR TITLE
PendingReadOp: improve logging for speculative read targets

### PR DESCRIPTION
**Pull request by athanatos**
And add more log messages around speculative read
to understand what bookies were tried, before attempting
speculative read.

(@bug W-3315760)
Signed-off-by: Venkateswararao Jujjuri (JV) <vjujjuri@salesforce.com>
[Fixed merge conflicts and checkstyle issues]
Signed-off-by: Samuel Just <sjust@salesforce.com>

### CODE REVIEW

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingReadOp.java [hunk 1]
```diff
@@ -254,9 +255,10 @@ public String toString() {
                 @Override
                 public Boolean call() throws Exception {
                     if (!isComplete() && null != maybeSendSpeculativeRead(heardFromHostsBitSet)) {
-                        if (LOG.isDebugEnabled()) {
-                            LOG.debug("Send speculative read for {}. Hosts heard are {}, ensemble is {}.",
-                                    this, heardFromHostsBitSet, ensemble);
+                        if (LOG.isInfoEnabled()) {
```
Comment (sijie) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingReadOp.java]: Are we sure changing this from debug to info? At a high load cluster, you might have 5-10% or even more traffic do speculative reads to ensure predictable latency. This change will result in producing a lot of unnecessary logging. 

May I ask you a question? How useful this logging helps you guys at production?

Comment (athanatos) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingReadOp.java]: It's been pretty useful for us.  Perhaps we should simply bump logging to debug in the few classes where we want it. @jvrao thoughts?

Comment (sijie) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingReadOp.java]: yeah. that would be ideal, you can configure your log4j properties to enable debug in PendingReadOp.

**Comment by athanatos** [comment]
@sijie Switched to debug.

**Pull request closed without merging**